Content.Shared/Xenoarchaeology/XenoArtifacts/ArtifactEffectPrototype.cs

```diff
@@ -33,4 +34,17 @@ public sealed class ArtifactEffectPrototype : IPrototype
 
     [DataField("effectHint")]
     public string? EffectHint;
+
+
+    /// <summary>
+    ///     Should this effect be restricted from artifacts with <see cref="ItemComponent"/>?
+    /// </summary>
+    [DataField("restrictItems")]
+    public bool RestrictItems = false;
+
+    /// <summary>
+    ///     Should this effect be restricted from artifacts that do not have <see cref="ItemComponent"/>?
+    /// </summary>
+    [DataField("restrictStructures")]
+    public bool RestrictStructures = false;
```

Comment:
EntityWhitelist.

Reply:
the base artifact components aren't going to really be dynamic enough for the entitywhitelist to be anything but this but i guess might as well